From c5216307e40ad37dda67cad4d077bc9e8b21ac5c Mon Sep 17 00:00:00 2001
From: Roy Li <rongqing.li@windriver.com>
Date: Sat, 22 Feb 2014 13:35:38 +0800
Subject: [PATCH] allow setrans to access /sys/fs/selinux

Upstream-Status: Pending

1. mcstransd failed to boot-up since the below permission is denied
statfs("/sys/fs/selinux", 0x7ffff2b80370) = -1 EACCES (Permission denied)

2. other programs can not connect to /run/setrans/.setrans-unix
type=AVC msg=audit(1393050297.936:25): avc:  denied  { connectto } for  pid=2055 comm="ls" path="/run/setrans/.setrans-unix" scontext=root:sysadm_r:sysadm_t:s0-s15:c0.c1023 tcontext=system_u:system_r:setrans_t:s15:c0.c1023 tclass=unix_stream_socket

Signed-off-by: Roy Li <rongqing.li@windriver.com>
---
 policy/modules/system/setrans.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/setrans.te b/policy/modules/system/setrans.te
index 3f50e54..050da2e 100644
--- a/policy/modules/system/setrans.te
+++ b/policy/modules/system/setrans.te
@@ -56,6 +56,8 @@ manage_files_pattern(setrans_t, setrans_var_run_t, setrans_var_run_t)
 manage_sock_files_pattern(setrans_t, setrans_var_run_t, setrans_var_run_t)
 files_pid_filetrans(setrans_t, setrans_var_run_t, { file dir })
 
+selinux_getattr_fs(setrans_t)
+mls_trusted_object(setrans_t)
 kernel_read_kernel_sysctls(setrans_t)
 kernel_read_system_state(setrans_t)
 
-- 
2.7.4

